### PR TITLE
First prototype If. MO part

### PR DESCRIPTION
**Pull request by evolosen**
Ticket: #51273

If prototype. MO Part.

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests - N/A
* [x]  Framework operation tests - N/A
* [x]  Transformation tests - N/A
* [ ]  Model Optimizer IR Reader check

Documentation:
* [ ]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: model-optimizer/extensions/front/tf/if_ext.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        # connect external input ports with body parameter nodes except input with condition
+        for idx in range(0, len(body_parameters)):
+            If.connect_body_input(if_node, not input_index, idx + 1, body_parameters[idx])
+
+        # connect body outputs with If operation output ports
+        for idx in range(len(body_results)):
+            If.connect_body_output(if_node, not input_index, idx, body_results[idx])
+
+        # # run function to parse body nodes attributes similar to the main graph
```
Comment (evolosen) [inline model-optimizer/extensions/front/tf/if_ext.py:93]: Need  delete #

File: model-optimizer/extensions/front/tf/if_ext.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    # fill the body graph
+    for attr_key in main_graph.graph.keys():
+        if attr_key != 'library':
+            then_graph.graph[attr_key] = copy.deepcopy(main_graph.graph[attr_key])
+        else:
+            # it is sufficient to have a link to the library
+            then_graph.graph['library'] = main_graph.graph['library']
+    if_node['then_graph'] = then_graph
+
+    # TODO refactor to have one function doing this
```
Comment (evolosen) [inline model-optimizer/extensions/front/tf/if_ext.py:51]: delete comment

File: model-optimizer/extensions/middle/IfConstCondition.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    output_port.disconnect()
+                    input_connection.add_destination(output_port)
+                if_node.graph.erase_node(external_node)  # erasing Result node
+
+        for in_port in if_node.in_ports().values():
+            in_port.disconnect()
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for if_node in graph.get_op_nodes(type='If'):
+            pass
```
Comment (evolosen) [inline model-optimizer/extensions/middle/IfConstCondition.py:77]: delete

File: model-optimizer/extensions/front/tf/if_ext.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        extract_node_attrs(body_graph, lambda node: tf_op_extractor(node, check_for_duplicates(tf_op_extractors)))
+    return cls.enabled
+
+
+class IfExtractor(FrontExtractorOp):
+    op = 'If'
+    enabled = True
+
+    @classmethod
+    def extract(cls, if_node: Node):
```
Comment (vgavrilo) [inline model-optimizer/extensions/front/tf/if_ext.py:103]: Maybe we need also extractor for ONNX If?

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/front/tf/if_ext.py (hunk 4 above)
Comment (evolosen) [reply model-optimizer/extensions/front/tf/if_ext.py:103]: no, we need if now for enabling tf2 center_net models. Full support if in next release

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Comment by achetver** [comment]
Have you checked MO IR Reader with any IR with this operation? You marked this point in description as done, but looks like this operation cannot be successfully restored and saved without additional changes inside MO IR Reader.

**Comment by lazarevevgeny** [comment]
Please, update the documentation for the TF with the list of supported operations with the comment when it is supported.

**Review by vgavrilo — APPROVED** [review]

File: model-optimizer/extensions/ops/If.py [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_type = if_node.in_port(input_port_id).get_data_type()
+                node.data_type = input_type
+                log.debug('Updated data type for the body node with name "{}" with value {}'
+                          .format(node.name, node.data_type))
+
+    @staticmethod
+    def updated_body_parameters_shape(if_node: Node, condition: bool):
```
Comment (sadolini) [inline model-optimizer/extensions/ops/If.py:97]: Why we need the second parameter? Could we just update shapes for both then and else?

**Pull request closed without merging**